include/ck_tile/host/kernel_launch.hpp

```diff
@@ -91,6 +91,29 @@ inline void remove_outliers(std::vector<float>& v)
     v.erase(std::remove_if(v.begin(), v.end(), [&](float f) { return f < lb || f > ub; }), v.end());
 }
 
+// Measure the preprocess time during the cold iterations
+template <typename TimerType>
+CK_TILE_HOST double
+preprocess_profiling_impl(TimerType timer, const stream_config& s, std::function<void()> preprocess)
+{
+    timer.start(s.stream_id_);
+    auto iter = max(s.cold_niters_, s.nrepeat_);
+    for(int i = 0; i < iter; i++)
+    {
+        if(preprocess)
+            preprocess();
+    }
+    timer.stop(s.stream_id_);
+
+    return timer.duration() / iter;
+}
+
+template <typename PreprocessFunc>
+CK_TILE_HOST float preprocess_profiling(const stream_config& s, PreprocessFunc preprocess)
+{
+    return preprocess_profiling_impl(gpu_timer{}, s, preprocess);
+}
+
 template <typename TimerType, typename CallablesFunc>
 CK_TILE_HOST double timing_loop_impl(TimerType timer,
                                      const stream_config& s,
```

Comment:
should preprocess() be a template parameter? We're getting a separate template instance for each call anyways